org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/ReferencesHandler.java

```diff
@@ -55,47 +62,100 @@ private IJavaSearchScope createSearchScope() throws JavaModelException {
+			if (elementToSearch == null) {
+				int offset = JsonRpcHelpers.toOffset(typeRoot.getBuffer(), param.getPosition().getLine(), param.getPosition().getCharacter());
+				elementToSearch = typeRoot.getElementAt(offset);
+			}
 			if (elementToSearch == null) {
 				return locations;
 			}
-
-			boolean includeClassFiles = preferenceManager.isClientSupportsClassFileContent();
-			SearchEngine engine = new SearchEngine();
-			SearchPattern pattern = SearchPattern.createPattern(elementToSearch, IJavaSearchConstants.REFERENCES);
-
-			engine.search(pattern, new SearchParticipant[] { SearchEngine.getDefaultSearchParticipant() }, createSearchScope(), new SearchRequestor() {
-
-				@Override
-				public void acceptSearchMatch(SearchMatch match) throws CoreException {
-					Object o = match.getElement();
-					if (o instanceof IJavaElement) {
-						IJavaElement element = (IJavaElement) o;
-						ICompilationUnit compilationUnit = (ICompilationUnit) element.getAncestor(IJavaElement.COMPILATION_UNIT);
-						Location location = null;
-						if (compilationUnit != null) {
-							location = JDTUtils.toLocation(compilationUnit, match.getOffset(), match.getLength());
-						} else if (includeClassFiles) {
-							IClassFile cf = (IClassFile) element.getAncestor(IJavaElement.CLASS_FILE);
-							if (cf != null && cf.getSourceRange() != null) {
-								location = JDTUtils.toLocation(cf, match.getOffset(), match.getLength());
-							}
-						}
-						if (location != null) {
-							locations.add(location);
+			search(elementToSearch, locations, monitor);
+			if (preferenceManager.getPreferences().isIncludeGetterSetter() && elementToSearch instanceof IField) { // IField
+				IField field = (IField) elementToSearch;
+				IMethod getter = GetterSetterUtil.getGetter(field);
+				if (getter != null) {
+					search(getter, locations, monitor);
+				}
+				IMethod setter = GetterSetterUtil.getSetter(field);
+				if (setter != null) {
+					search(setter, locations, monitor);
+				}
+				String builderName = getBuilderName(field);
+				IType builder = field.getJavaProject().findType(builderName);
+				if (builder != null) {
+					String fieldSignature = field.getTypeSignature();
+					for (IMethod method : builder.getMethods()) {
+						String[] parameters = method.getParameterTypes();
+						if (parameters.length == 1 && field.getElementName().equals(method.getElementName()) && fieldSignature.equals(parameters[0])) {
+							search(method, locations, monitor);
 						}
 					}
 				}
-			}, monitor);
-
+			}
 		} catch (CoreException e) {
 			JavaLanguageServerPlugin.logException("Find references failure ", e);
 		}
 		return locations;
 	}
 
+	private String getBuilderName(IField field) {
+		IType declaringType = field.getDeclaringType();
+		IAnnotation annotation = declaringType.getAnnotation("Builder");
+		if (annotation == null || !annotation.exists()) {
+			annotation = declaringType.getAnnotation("lombok.Builder");
+		}
```

Comment:
This may eventually have to evolve to support other builders (eg. https://immutables.github.io/ ) but I think it should be fine for now.